www/js/lib/uiUtil.js

```diff
@@ -465,80 +465,159 @@ function isElementInView (el, fully) {
+    sectionToShow.classList.add('slideIn_L');
+    setTimeout(function () {
+        sectionToShow.style.display = '';
+        // sectionToShow.classList.remove('slideIn_L');
+    }, 300);
+
+    sectionToHide.classList.add('slideOut_L');
+    setTimeout(function () {
+        sectionToHide.style.display = 'none';
+        // sectionToHide.classList.remove('slideOut_L');
+    }, 300);
 }
 
 /**
  * Adds the slide animation between different sections
  *
- * @param {String} section It takes the name of the section to which the animation is to be added
+ * @param {HTMLElement} sectionToShow Element which is gonna be slide in from right (show)
+ * @param {HTMLElement} sectionToHide Element which is gonna be slide to the right (hide)
+ * @returns {void}
+ */
+function slideToRight (sectionToShow, sectionToHide) {
+    sectionToHide.classList.add('slideOut_R');
+    setTimeout(function () {
+        sectionToHide.style.display = 'none';
+        // sectionToHide.classList.remove('slideOut_R');
+    }, 300);
+
+    sectionToShow.classList.add('slideIn_R');
+    setTimeout(function () {
+        sectionToShow.style.display = '';
+        // sectionToShow.classList.remove('slideIn_R');
+    }, 300);
+}
+
+/**
+ * Returns the name of the section which is currently visible
+ * @returns {String} The name of the section which is currently visible
+ */
+function fromSection () {
+    const isConfigPageVisible = !$('#configuration').is(':hidden');
+    const isAboutPageVisible = !$('#about').is(':hidden');
+    const isArticlePageVisible = !$('#articleContent').is(':hidden');
+
+    if (isConfigPageVisible) return 'config';
+    else if (isAboutPageVisible) return 'about';
+    else if (isArticlePageVisible) return 'home';
+}
+/**
+ * Adds the slide animation between different sections
  *
+ * @param {String} toSection It takes the name of the section to which the animation is to be added
+ * @param {Boolean} isAnimationRequired To enable or disable the animation
+ * @returns {void}
  */
-function applyAnimationToSection (section) {
-    if (section === 'home') {
-        if (!$('#configuration').is(':hidden')) {
-            document.getElementById('configuration').classList.add('slideOut_R');
-            setTimeout(function () {
-                document.getElementById('configuration').style.display = 'none';
-            }, 300);
-        }
-        if (!$('#about').is(':hidden')) {
-            document.getElementById('about').classList.add('slideOut_R');
-            setTimeout(function () {
-                document.getElementById('about').style.display = 'none';
-            }, 300);
+function tabTransitionToSection (toSection, isAnimationRequired = false) {
+    // all the references of the sections/tabs
+    const config = document.getElementById('configuration');
+    const about = document.getElementById('about');
+    const home = document.getElementById('articleContent');
+
+    const isZimLoaded = true // get from settings or something
```

Comment:
Could you tell me how I check if a Zim file is loaded
do I directly use DOM and check if any file is uploaded `document.getElementById("archiveFiles").files.length`
or is there any other variable (I didn't found anything similar)

Reply:
There is the object `selectedArchive` which represents a loaded ZIM archive. It has a method `isReady` (or something like that).